### DISCUSSION TITLE
Decomposing `auto shipit` to component commands

### DISCUSSION THREAD

**Discussion by jwodder** [intuit/auto#2072] [Q&A]
We have an auto configuration that currently uses the `exec` plugin to run an `afterChangelog` hook.  However, due to [the E2BIG problem](https://github.com/intuit/auto/issues/1294), we have to abandon this configuration and are looking into alternative ways to get auto to do what we want.  One way would be to replace the `auto shipit` command with an appropriate series of `auto` subcommands with the `afterChangelog` commands interspersed.  Is the following Bash script:

```bash
if [ -n "$(auto version)" ]
then auto changelog
     ### Insert afterChangelog commands here
     auto release
fi
```

the same as running `auto shipit` with an `afterChangelog` hook configured via `exec`, particularly in the case where `onlyPublishWithReleaseLabel` is set,`git-tag` is listed in `plugins`, and the script is run from the base branch?

**Comment by laughedelic** [comment]
Sounds similar to https://github.com/intuit/auto/discussions/1965, but maybe not exactly the same. Just linking for the context.
